parser_library/src/workspaces/workspace.cpp

```diff
@@ -448,19 +455,19 @@ workspace_file_info workspace::parse_file(
 
     // TODO: apparently just opening a file without changing it triggers reparse
 
-    if (processor_file_compoments* this_file = nullptr; file_content_status == open_file_result::changed_content
+    if (processor_file_compoments* this_proc_file = nullptr; file_content_status == open_file_result::changed_content
         || file_content_status == open_file_result::changed_lsp
-            && !((this_file = find_processor_file_impl(file_location)) != nullptr
-                && this_file->m_processor_file->has_lsp_info()))
+            && (this_proc_file = find_processor_file_impl(file_location)) != nullptr)
```

Comment:
This will certainly cause reparsing of dependencies, just because you open them in VSCode without content actually changing...

Reply:
Didn't realize that. I'll add a test for this

Reply:
Test is still wip